packages/server/utils/ServerHealthChecker.ts

```diff
@@ -8,45 +8,47 @@ import sendToSentry from './sendToSentry'
 
 const REDIS_URL = process.env.REDIS_URL!
 const SERVER_ID = process.env.SERVER_ID!
+const INSTANCE_ID = `${SERVER_ID}:${process.pid}`
 
-export default class ServerHealthChecker {
-  publisher = new Redis(REDIS_URL, {connectionName: 'serverHealth_pub'})
-  subscriber = new Redis(REDIS_URL, {connectionName: 'serverHealth_sub'})
-  pendingPongs: null | Set<string> = null
-  constructor() {
+class ServerHealthChecker {
+  private publisher = new Redis(REDIS_URL, {connectionName: 'serverHealth_pub'})
+  private subscriber = new Redis(REDIS_URL, {connectionName: 'serverHealth_sub'})
+  private remoteSocketServers: null | string[] = null
+  private joinPoolPromise: null | Promise<void> = null
+  private async joinPool() {
+    if (this.joinPoolPromise) throw new Error('joinPool can only be called once')
```

Comment:
-1 I don't like that this read and check of `this.joinPoolPromise` is inside this call, whereas the write is outside. I think the check and assignment should either be completely in `joinPool` or completely in `getLivingServers`.
I think cleanest would be just putting all the logic inside `joinPool`
```ts
private async joinPool() {
  if (!this.joinPoolPromise) {
    this.joinPoolPromise = async () => {
      await this.subscriber.subscribe('socketServerPing', `socketServerPong:${INSTANCE_ID}`)
      ...
    }()
  }
  return this.joinPoolPromise
}
```

Reply:
💯 that is cleaner